### PR TITLE
CI Cache Optimization

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Changed the CI to make better and more use of various caching systems in github.
CI now uses python 3.9.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
a comment that should not retrigger the CI

**Comment by FluffyGhoster** [comment]
g!rerun

**Comment by FluffyGhoster** [comment]
!review

File: .github/workflows/byond.yml [hunk 1]
```diff
@@ -48,13 +48,25 @@ jobs:
...
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          cache-dependency-path: tgui/yarn.lock
+          node-version: 14.16.1
+          cache: 'yarn'
```
Comment (Arrow768) [inline .github/workflows/byond.yml:61]: You are specifying the versions in this file in addition to the dependencies file

**Review by Arrow768 — COMMENTED** [review]
Imho it would be preferable if the versions are loaded from the dependencies file only.

**Review by Alberyk — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**